### PR TITLE
Add support for raw lucene date range queries

### PR DESCRIPTION
**Pull request by mad**
fix for #1614

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java [hunk 1]
```diff
@@ -67,21 +73,21 @@ protected Query getRangeQuery(final String field, final String start, final Stri
     }
 
     @Override
-    protected Query newTermQuery(Term t) {
```
Comment (porunov) [inline janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java]: Why you removed `newTermQuery`?

Comment (mad) [reply janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java]: By call hierarchy newTermQuery invoked from newFieldQuery

**Comment by mad** [comment]
I also add `setAllowLeadingWildcard` by default

QueryParser may change frequently, may be add user supplied `RawQuery.parameter` for determine what is it parser will be used?

**Comment by porunov** [comment]
> I also add setAllowLeadingWildcard by default
> 
> QueryParser may change frequently, may be add user supplied RawQuery.parameter for determine what is it parser will be used?

Why did you set it to `true` by default?
I think if you need to setup it to true then you should set it where it is needed. Currently in `LuceneIndex`.
I.e.
```
NumericTranslationQueryParser queryParser = new NumericTranslationQueryParser(information.get(query.getStore()), "_all", analyzer);
queryParser.setAllowLeadingWildcard(true);
q = queryParser.parse(query.getQuery());
```

Also, I think we could cache `NumericTranslationQueryParser` for each `storeRetriever` in some map to optimize `query` (i.e. to not create a new `NumericTranslationQueryParser` for each query) but it is out of the scope of this PR.

**Comment by porunov** [comment]
@mad can you rebase please? Due to merging of #1655 there are conflicts now

**Comment by mad** [comment]
@porunov done, also enable luceneStyleQueries here https://github.com/JanusGraph/janusgraph/pull/1615/files#diff-afebec2aac510e2f96677ccb01775934L49

**Comment by porunov** [comment]
Merging by following lazy consensus

**Pull request merged**